integrations/gcp/gcp_core/search/resource_searches.py

```diff
@@ -25,169 +24,186 @@
+                ):
+                    yield assets
+
+            except PermissionDenied:
+                logger.error(
+                    f"Service account doesn't have permissions to search all resources within project {project_name} for kind {asset_type}"
+                )
+            except NotFound:
+                logger.debug(
+                    f"Couldn't perform search_all_resources on project {project_name} since it's deleted."
+                )
+            else:
+                logger.info(
+                    f"Successfully searched all resources within project {project_name}"
                 )
-            )
-            async for paginated_response in paginated_responses.pages:
-                raw_assets = parse_protobuf_messages(paginated_response.results)
-                assets = typing.cast(list[AssetData], raw_assets)
-                if assets:
-                    logger.info(f"Found {len(assets)} {asset_type}'s")
-                    latest_resources = []
-                    for asset in assets:
-                        latest_resource = parse_latest_resource_from_asset(asset)
-                        latest_resource[EXTRA_PROJECT_FIELD] = project
-                        latest_resources.append(latest_resource)
-                    yield latest_resources
-        except PermissionDenied as e:
-            logger.exception(
-                f"Couldn't access the API Cloud Assets to get kind {asset_type}. Please set cloudasset.assets.searchAllResources permissions for project {project_name}"
-            )
-            raise e
 
 
 async def list_all_topics_per_project(
-    project: dict[str, Any],
+    project: dict[str, Any], **kwargs: Any
 ) -> ASYNC_GENERATOR_RESYNC_TYPE:
-    """
-    This lists all Topics under a certain project.
-    The Topics are handled specifically due to lacks of data in the asset itselfwithin the asset inventory - e.g. some properties missing.
-    The listing is being done via the PublisherAsyncClient, ignoring state in assets inventory
-    """
-    project_name = project["name"]
-    logger.info(
-        f"Searching all {AssetTypesWithSpecialHandling.TOPIC}'s in project {project_name}"
-    )
     async with PublisherAsyncClient() as async_publisher_client:
-        try:
-            list_topics_pagers = await async_publisher_client.list_topics(
-                project=project_name, retry=retry_policy
-            )
-            async for paginated_response in list_topics_pagers.pages:
-                topics = parse_protobuf_messages(paginated_response.topics)
-                if topics:
-                    logger.info(
-                        f"Found {len(topics)} {AssetTypesWithSpecialHandling.TOPIC}'s"
-                    )
+        project_name = project["name"]
+        logger.info(
+            f"Searching all {AssetTypesWithSpecialHandling.TOPIC}'s in project {project_name}"
+        )
+        async with kwargs["topic_semaphore"]:
```

Comment:
Do we need a semaphore or a aiolimiter here? I believe we need a aiolimiter, correct me if im wrong

Reply:
You are absolutely right, I'll wouldn't be necessary to apply a semaphore on querying the topics kind for at the moment